src/index.ts

```diff
@@ -63,37 +574,636 @@ const dmMachine = createMachine(
+                  });
+                },
+                on: { SPEAK_COMPLETE: "GettingPurpose" },
+              },
+              GettingPurpose: {
+                entry: listen(),
+                on: {
+                  RECOGNISED: {
+                    target: "CompleteMeal", //change state to say name. "so <name> what can i do for you today?"
+                    actions: [
+                      ({ event }) => console.log(event),
+                      assign({
+                        purpose: ({ event }) => {
+                          const u = event.value[0].utterance.toLowerCase().replace(/\.$/g, "").replace("for ", "")
+                          if (u === "no" || u === "no, thank you" || u === "no thank you") {
+                            return "no preference"
+                          }
+                          return u
+                        },
+                      }),
+                    ],
                   },
                 },
               },
-              Repeat: {
+              // when user gives only when and purpose => state for filling out "what"
+              MissingWhat: {
                 entry: ({ context }) => {
                   context.spstRef.send({
                     type: "SPEAK",
-                    value: { utterance: context.lastResult[0].utterance },
+                    value: { utterance: `Could you provide me with what would you like to eat, pizza, donughts, burger?` },
                   });
                 },
-                on: { SPEAK_COMPLETE: "Ask" },
+                on: { SPEAK_COMPLETE: "GettingWhat" },
+              },
+              GettingWhat: {
+                entry: listen(),
+                on: {
+                  RECOGNISED: {
+                    target: "CompleteMeal", //change state to say name. "so <name> what can i do for you today?"
+                    actions: [
+                      ({ event }) => console.log(event),
+                      assign({
+                        what: ({ event }) => {
+                          const u = event.value[0].utterance.toLowerCase().replace(/\.$/g, "").replace("i would like to eat ", "").replace("i want to eat ", "").replace("i'm craving for ","").replace("i want a ").replace(" sounds nice","")
+                          return u
+                        },
+                      }),
+                    ],
+                  },
+                },
+              },
+              //missing BOTH what and purpose events
+              onlyHaveWhen: {
+                entry: ({ context }) => {
+                  context.spstRef.send({
+                    type: "SPEAK",
+                    value: { utterance: `Could you provide me with what would you like to eat and if there is a specific preference or occasion? For example you can say: vegan pizza or italian food for a date.` },
```

Comment:
You sort of fill two slots at once, which is good, but not what was expected from form-filling dialogue.